### ISSUE TITLE
ad_hoc_data() should handle error gracefully when n > 3

### ISSUE THREAD

**Issue by omarshehab** [Qiskit/qiskit-machine-learning#221]
### Information

- **Qiskit Machine Learning version**: 0.2.0
- **Python version**: 3.9.5 (default, May 18 2021, 12:31:01) [Clang 10.0.0 ]
- **Operating system**: Darwin

### What is the current behavior?
When `n > 3`, `ad_hoc_data()` generates the following runtime error.

```
~/opt/anaconda3/envs/py39/lib/python3.9/site-packages/qiskit_machine_learning/datasets/ad_hoc.py in ad_hoc_data(training_size, test_size, n, gap, plot_data, one_hot, include_sample_total)
     51     # interactions = np.transpose(np.array([[1, 0], [0, 1], [1, 1]]))
     52 
---> 53     steps = 2 * np.pi / count
     54 
     55     # sx = np.array([[0, 1], [1, 0]])

ZeroDivisionError: float division by zero
```

### Steps to reproduce the problem

Set `n > 3` when invoking `ad_hoc_data()`.

### What is the expected behavior?
It should just return some kind of error value.

**Comment by woodsp-ibm** [comment]
I think it is setup for n or 2 or 3 only. In which case it would be better if any other value was rejected with a suitable error message.

Also this ought to be in the documentation for the dataset. I see in general the datasets have some very minimal docstring that does not document the args. So things could be improved in this area as well.

**Comment by adekusar-drl** [comment]
Fixed in #287
